static/app/views/dashboardsV2/detail.tsx

```diff
@@ -265,39 +265,46 @@ class DashboardDetail extends Component<Props, State> {
     });
   };
 
-  onAddWidget = () => {
+  handleAddLibraryWidgets = (widgets: Widget[]) => {
     const {organization, dashboard, api, onDashboardUpdate, location} = this.props;
+    const {dashboardState} = this.state;
+    const modifiedDashboard = {
+      ...cloneDashboard(dashboard),
+      widgets: widgets.map(assignTempId),
+    };
+    this.setState({modifiedDashboard});
+    if (dashboardState === DashboardState.CREATE) {
+      return;
+    }
+    updateDashboard(api, organization.slug, modifiedDashboard).then(
```

Comment:
Since this actually triggers the update, a user can't cancel and revert their changes after adding widgets. Is this okay?

Reply:
DANG GOOD CATCH

Reply:
Addressed in 2f8c6d749a3f696b595dc92fc577e3f6ca54146d returns early if dashboard is in edit/create mode since the update is handled onSave